src/app/components/Filters/FieldsetDate.jsx

```diff
@@ -1,27 +1,100 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import Actions from '../../actions/Actions.js';
+import {
+  extend as _extend,
+} from 'underscore';
+
+class FieldsetDate extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dateAfter: '0',
+      dateBefore: '0',
+      selectedFacets: this.props.selectedFacets,
+    };
+
+    this.inputChange = this.inputChange.bind(this);
+  }
+
+  /**
+   * inputChange()
+   * Updates the states and the store based on the input values.
+   *
+   */
+  inputChange(e) {
+    const value = (e.target.value).replace(/[a-zA-Z]/g, '');
```

Comment:
From my comment below, make the input a type `number` so you don't have to do this regex. Also, I can currently put "3das" as a value and that shouldn't be allowed.

Reply:
I just had a quick chat with @wlla , it sounds like there is a goal to allow special characters such as `195*` to indicate the range from `1950` to `1959`?

Reply:
That'd be cool to do in the future, but if you check the Acceptance Criteria in the Jira ticket, it says 4 digits. Let's do the simple case first that's in scope for this sprint and update later.